src/business_logic/transaction/objects/internal_declare.rs

```diff
@@ -0,0 +1,428 @@
+use std::collections::HashMap;
+
+use felt::{felt_str, Felt};
+use num_traits::Zero;
+
+use crate::{
+    business_logic::{
+        execution::{
+            error::ExecutionError,
+            execution_entry_point::ExecutionEntryPoint,
+            objects::{CallInfo, TransactionExecutionContext, TransactionExecutionInfo},
+        },
+        fact_state::state::ExecutionResourcesManager,
+        state::state_api::{State, StateReader},
+        transaction::{
+            error::TransactionError,
+            fee::{calculate_tx_fee, execute_fee_transfer, FeeInfo},
+        },
+    },
+    core::{
+        contract_address::starknet_contract_address::compute_class_hash,
+        transaction_hash::starknet_transaction_hash::calculate_declare_transaction_hash,
+    },
+    definitions::{general_config::StarknetGeneralConfig, transaction_type::TransactionType},
+    services::api::contract_class::{ContractClass, EntryPointType},
+    utils::{calculate_tx_resources, felt_to_hash, verify_no_calls_to_other_contracts, Address},
+};
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+///  Represents an internal transaction in the StarkNet network that is a declaration of a Cairo
+///  contract class.
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+pub(crate) struct InternalDeclare {
+    pub(crate) class_hash: [u8; 32],
+    pub(crate) sender_address: Address,
+    pub(crate) tx_type: TransactionType,
+    pub(crate) validate_entry_point_selector: Felt,
+    pub(crate) version: u64,
+    pub(crate) max_fee: u64,
+    pub(crate) signature: Vec<Felt>,
+    pub(crate) nonce: Felt,
+    pub(crate) hash_value: Felt,
+}
+
+// ------------------------------------------------------------
```

Comment:
I think this comment is redundant

Reply:
I think it helps to delimited the functions but may be deleted.